### PR TITLE
fix: :safety_vest: Safety checks on data implemented

### PR DESCRIPTION
**Pull request by CharelMarais**
Implemented safety check for errors on variables

### CODE REVIEW

File: src/components/ErrorComponent.tsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export function ErrorComponent() {
+  return (
+    <div className="pt-14 flex justify-center items-center flex-col h-full text-neutral-200">
+      <h2 className="text-6xl font-semibold">404</h2>
```
Comment (MicNow27) [inline src/components/ErrorComponent.tsx:4]: The 404 heading doesn't show on wider screens

File: src/components/ErrorComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+      <img className="-my-4" src="/src/assets/img/yamcha.png" alt="" />
+      <p>Sorry, an unexpected error has occurred.</p>
```
Comment (MicNow27) [inline src/components/ErrorComponent.tsx:6]: On wider screens, this displays below "the fold". User must scroll to see it.

File: src/components/ChapterFeed.tsx [hunk 3]
```diff
@@ -5,33 +5,41 @@ import { IMangaChapterFeedProp } from "../props/mangaProps";
...
-              <p className="cursor-pointer text-neutral-400 hover:bg-stone-700">
-                Chapter: {data.attributes.chapter} Pages:{" "}
-                {data.attributes.pages}
-              </p>
-            </Link>
-          );
-        })}
-    </div>
+    <>
+      {isSuccess && (
```
Comment (MicNow27) [inline src/components/ChapterFeed.tsx:15]: It seems that even if `isSuccess`, `data` still can still come back as empty, then `No English Translated Chapters` is not displayed.

File: src/components/ErrorComponent.tsx (hunk 1 above)
Comment (CharelMarais) [reply src/components/ErrorComponent.tsx:4]: I am working on this with my next branch for responsive design

File: src/components/ErrorComponent.tsx (hunk 2 above)
Comment (CharelMarais) [reply src/components/ErrorComponent.tsx:6]: A fixed size for the image will be implemented on larger screens

File: src/components/ChapterFeed.tsx (hunk 3 above)
Comment (CharelMarais) [reply src/components/ChapterFeed.tsx:15]: That is correct that is because I am looking only for the en translated manga chapters

**Pull request merged**